### PR TITLE
Validating CIDR of network conf file at install-cni.sh

### PR DESCRIPTION
**Pull request by AjeyHiremath**

### CODE REVIEW

File: scripts/install-cni.sh [hunk 1]
```diff
@@ -1,7 +1,23 @@
...
-
+#! /bin/bash
 set -e -x;
 
+checksubnet(){
+	subnt=`jq '.ipam.subnet' /install/config/$CONF_FILE_NAME`
+    temp="${subnt%\"}"
+    temp="${temp#\"}"
+	if [ -z ` egrep '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/([0-9]|[1-2][0-9]|3[0-2]))$' <<< $temp` ]; then
+		echo "invalid CIDR mention in ${CONF_FILE_NAME}"
```
Comment (chinmayb) [inline scripts/install-cni.sh]: change it to "Invalid CIDR mentioned in the conf file ${CONF}"

File: scripts/install-cni.sh [hunk 2]
```diff
@@ -16,12 +32,16 @@ fi;
...
    ATIME=`stat -c %Z /install/config/..data/$CONF_FILE_NAME`
 
    if [[ "$ATIME" != "$LTIME" ]]
-   then
-    cp -f /install/config/$CONF_FILE_NAME /host/etc/cni/net.d/;
-    LTIME=$ATIME
+   then       
+	    checksubnet 0        
+        if [ $? -eq 0 ]; then
+			cp -f /install/config/$CONF_FILE_NAME /host/etc/cni/net.d/;
```
Comment (chinmayb) [inline scripts/install-cni.sh]: indent properly

File: scripts/install-cni.sh [hunk 3]
```diff
@@ -16,12 +32,16 @@ fi;
 LTIME=`stat -c %Z /install/config/..data/$CONF_FILE_NAME`
 while true
 do
+    
    ATIME=`stat -c %Z /install/config/..data/$CONF_FILE_NAME`
```
Comment (chinmayb) [inline scripts/install-cni.sh]: /install/config is used frequently can be used as a variable.

File: Dockerfile-cni-plugin-installer [hunk 4]
```diff
@@ -13,6 +13,9 @@ FROM alpine:3.5
 ENV SRC=/go/src/github.com/infobloxopen/cni-infoblox
 COPY --from=builder ${SRC}/bin/infoblox-plugin /install/bin/infoblox
 
+RUN apk add --no-cache bash
```
[the remark this answers is not in the document]
Comment (AjeyHiremath) [reply Dockerfile-cni-plugin-installer]: done

File: scripts/install-cni.sh [hunk 5]
```diff
@@ -1,7 +1,23 @@
-#!/bin/sh
-
+#! /bin/bash
 set -e -x;
```
[the remark this answers is not in the document]
Comment (chinmayb) [reply scripts/install-cni.sh]: lets not check in the final version. what say @jkraj  ?

File: Dockerfile-cni-plugin-installer [hunk 6]
```diff
@@ -13,6 +13,8 @@ FROM alpine:3.5
 ENV SRC=/go/src/github.com/infobloxopen/cni-infoblox
 COPY --from=builder ${SRC}/bin/infoblox-plugin /install/bin/infoblox
 
+RUN apk add --no-cache bash jq
+
 ADD scripts/install-cni.sh /install/install-cni.sh
 RUN chmod +x /install/bin/infoblox
 RUN chmod +x /install/install-cni.sh
```
Comment (chinmayb) [inline Dockerfile-cni-plugin-installer]: i was implying all these run commands can be in single line with &

File: scripts/install-cni.sh [hunk 7]
```diff
@@ -1,27 +1,57 @@
...
+CONF_FILE_PATH_HOST="/host/etc/cni/net.d/"
+BIN_FILE_PATH_HOST="/host/opt/cni/bin/"
+CONF_FILE="${CONF_FILE_PATH}/${CONF_FILE_NAME}"
+
+# Validating CIDR from network configuration file
+checksubnet(){
+	#reads subnet from network configuration file
+	subnt=`jq '.ipam.subnet' $CONF_FILE`
+    temp="${subnt%\"}"
+    temp="${temp#\"}"
```
Comment (chinmayb) [inline scripts/install-cni.sh]: why in two lines ?

File: scripts/install-cni.sh [hunk 8]
```diff
@@ -1,27 +1,57 @@
...
-    cp -f /install/config/* /host/etc/cni/net.d/;
-    echo "Wrote netconf to /host/etc/cni/net.d/";
+if [ -w "$CONF_FILE_PATH_HOST" ]; then
+    cp -f $CONF_FILE_PATH/* $CONF_FILE_PATH_HOST;
+    echo "Wrote netconf to $CONF_FILE_PATH_HOST";
 fi;
 
-# Polls for file change at /install/config/..data/$CONF_FILE_NAME
-LTIME=`stat -c %Z /install/config/..data/$CONF_FILE_NAME`
+# Polls for file change at $CONF_FILE_PATH/..data/$CONF_FILE_NAME
```
Comment (chinmayb) [inline scripts/install-cni.sh]: search and replace :D. Make it readable.

File: scripts/install-cni.sh [hunk 9]
```diff
@@ -1,27 +1,57 @@
...
-    cp -f /install/config/$CONF_FILE_NAME /host/etc/cni/net.d/;
-    LTIME=$ATIME
+    if [[ "$CURRENT_MODIFIED_TIME" != "$LAST_MODIFIED_TIME" ]]
+    then
+	    
+		checksubnet 0
+		     
+        if [ $? -eq 0 ]; then
+			cp -f $CONF_FILE_PATH/$CONF_FILE_NAME $CONF_FILE_PATH_HOST;
+	        LAST_MODIFIED_TIME=$CURRENT_MODIFIED_TIME
```
Comment (chinmayb) [inline scripts/install-cni.sh]: lets indent it properly.

File: scripts/install-cni.sh [hunk 10]
```diff
@@ -1,27 +1,57 @@
...
+    
+    CURRENT_MODIFIED_TIME=`stat -c %Z $CONF_FILE_PATH/..data/$CONF_FILE_NAME`
 
-   if [[ "$ATIME" != "$LTIME" ]]
-   then
-    cp -f /install/config/$CONF_FILE_NAME /host/etc/cni/net.d/;
-    LTIME=$ATIME
+    if [[ "$CURRENT_MODIFIED_TIME" != "$LAST_MODIFIED_TIME" ]]
+    then
+
```
Comment (chinmayb) [inline scripts/install-cni.sh]: Good to include some debug messages here

File: scripts/install-cni.sh [hunk 11]
```diff
@@ -1,27 +1,57 @@
...
-
+#! /bin/bash
 set -e -x;
 
-if [ -w "/host/opt/cni/bin/" ]; then
-    cp -f /install/bin/* /host/opt/cni/bin/;
-    echo "Wrote CNI binaries to /host/opt/cni/bin/";
+CONF_FILE_PATH="/install/config"
+BIN_FILE_PATH="/install/bin"
+CONF_FILE_PATH_HOST="/host/etc/cni/net.d/"
```
Comment (chinmayb) [inline scripts/install-cni.sh]: Im not sure about this, but CONF_FILE_PATH_SRC and CONF_FILE_PATH_DEST will be more accurate ? @jkraj

File: scripts/install-cni.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (jkraj) [reply scripts/install-cni.sh]: I feel it would be good to debug. So lets stay.

File: scripts/install-cni.sh (hunk 11 above)
Comment (AjeyHiremath) [reply scripts/install-cni.sh]: done

File: scripts/install-cni.sh (hunk 7 above)
Comment (AjeyHiremath) [reply scripts/install-cni.sh]: done

File: scripts/install-cni.sh (hunk 8 above)
Comment (AjeyHiremath) [reply scripts/install-cni.sh]: done

File: scripts/install-cni.sh (hunk 10 above)
Comment (AjeyHiremath) [reply scripts/install-cni.sh]: done

File: scripts/install-cni.sh (hunk 9 above)
Comment (AjeyHiremath) [reply scripts/install-cni.sh]: done

File: scripts/install-cni.sh [hunk 12]
```diff
@@ -1,57 +1,51 @@
...
-		     
+do    
+    CURRENT_MODIFIED_TIME=$(stat -c %Z ${CONF_FILE})
+    if [[ "$CURRENT_MODIFIED_TIME" != "$LAST_MODIFIED_TIME" ]]; then	    
+		checksubnet 0		     
         if [ $? -eq 0 ]; then
-			cp -f $CONF_FILE_PATH/$CONF_FILE_NAME $CONF_FILE_PATH_HOST;
-	        LAST_MODIFIED_TIME=$CURRENT_MODIFIED_TIME
+		    echo "Network conf is modified so changing..."
+            cp -f ${CONF_FILE_CONTAINER_PATH}/${CONF_FILE_NAME} ${CONF_FILE_HOST_PATH};
```
Comment (jkraj) [inline scripts/install-cni.sh]: correct the indentation

File: scripts/install-cni.sh [hunk 13]
```diff
@@ -1,57 +1,51 @@
...
-		echo "Invalid CIDR mentioned in the conf file ${CONF_FILE_NAME}"
-		if [ $1 -eq 1 ]; then # exit the script
-			exit 255
-		fi
-		return 1
-	else
-		return 0		
+checksubnet() {
+
+  	SUBNET=$(jq '.ipam.subnet' ${CONF_FILE} | tr -d \")
```
Comment (jkraj) [inline scripts/install-cni.sh]: correct the indentation

Comment (AjeyHiremath) [reply scripts/install-cni.sh]: done

File: scripts/install-cni.sh (hunk 12 above)
Comment (AjeyHiremath) [reply scripts/install-cni.sh]: done

File: scripts/install-cni.sh [hunk 14]
```diff
@@ -1,27 +1,53 @@
...
+        return 0
+    fi
+    if [ -z $(egrep '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/([0-9]|[1-2][0-9]|3[0-2]))$' <<< $SUBNET) ]; then
+        echo "Invalid CIDR mentioned in the conf file ${CONF_FILE_NAME}"
+  	    if [ $1 -eq 1 ]; then # exit the script
+  	        exit 255
+  	    fi
+        return 1
+    fi
+        return 0
```
Comment (jkraj) [inline scripts/install-cni.sh]: indentation Ajey.

There is no else block. why 'return 0' is starting from there

Comment (AjeyHiremath) [reply scripts/install-cni.sh]: done

Comment (chinmayb) [reply scripts/install-cni.sh]: yes,  lets not ignore indentation please. makes it more readable.

**Pull request merged**